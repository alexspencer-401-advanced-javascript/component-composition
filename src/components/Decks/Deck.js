import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const Deck = ({ items, title }) => {
  const propElements = items.map(prop => (
    <Card key={prop._id} {...prop} />
  ));

  return (
    <>
      <header>
        <h2>{title}</h2>
      </header>
      <ul>
        {propElements}
      </ul>
    </>
  );
};

Deck.propTypes = {
  title: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string,
    name: PropTypes.string,
    image: PropTypes.string,
    gifLink: PropTypes.string
  })),
};

export default Deck;
