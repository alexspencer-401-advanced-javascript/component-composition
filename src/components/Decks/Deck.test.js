import React from 'react';
import { shallow } from 'enzyme';
import Deck from './Deck';

describe('Deck component', () => {
  it('renders Deck', () => {
    const wrapper = shallow(<Deck items={[
      {
        '_id': '5da237699734fdcb7bef8f51',
        'name': 'Arnold Shortman',
        'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333'
      }
    ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
