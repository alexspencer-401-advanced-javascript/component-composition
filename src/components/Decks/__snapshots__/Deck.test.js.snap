// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck component renders Deck 1`] = `
<Fragment>
  <header>
    <h2 />
  </header>
  <ul>
    <Card
      _id="5da237699734fdcb7bef8f51"
      image="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
      key="5da237699734fdcb7bef8f51"
      name="Arnold Shortman"
    />
  </ul>
</Fragment>
`;
